Library/Homebrew/commands.rb

```diff
@@ -41,6 +41,15 @@ def self.valid_internal_dev_cmd?(cmd)
     require?(HOMEBREW_DEV_CMD_PATH/cmd)
   end
 
+  def self.valid_ruby_cmd?(cmd)
+    if (valid_internal_cmd?(cmd) || valid_internal_dev_cmd?(cmd) || external_ruby_v2_cmd_path(cmd)) &&
+       (command = Homebrew::AbstractCommand.command(cmd))
+      command.ruby_cmd?
+    else
+      false
+    end
```

Comment:
```suggestion
    (valid_internal_cmd?(cmd) || valid_internal_dev_cmd?(cmd) || external_ruby_v2_cmd_path(cmd)) &&
       Homebrew::AbstractCommand.command(cmd).ruby_cmd?
```
Any reason not to do this? Or, if you're worried about readability:
```suggestion
    valid_cmd = (valid_internal_cmd?(cmd) || valid_internal_dev_cmd?(cmd) || external_ruby_v2_cmd_path(cmd))
    ruby_cmd = Homebrew::AbstractCommand.command(cmd).ruby_cmd?
    
    valid_cmd && ruby_cmd
```